profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/features/DataCenterApi.java

```
@@ -0,0 +1,156 @@
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PATCH;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+@Path("/datacenters")
+@RequestFilters(BasicAuthentication.class)
+public interface DataCenterApi extends Closeable {
+
+   @Named("datacenter:list")
+   @GET
+   @Produces("application/vnd.profitbricks.resource+json")
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<DataCenter> list();
+   
+   @Named("datacenter:list")
+   @GET
+   @Produces("application/vnd.profitbricks.resource+json")
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<DataCenter> list(DepthOptions options);
+   
+   /**
+    * @param identifier Data Center identifier
+    * @return Returns information about an existing virtual data center's state and configuration or <code>null</code>
+    * if it doesn't exist.
+    */
+   @Named("datacenter:get")
+   @GET   
+   @Path("/{id}")
+   @ResponseParser(DataCenterParser.class)
+   @Produces("application/vnd.profitbricks.resource+json")
+   @Fallback(NullOnNotFoundOr404.class)
+   DataCenter getDataCenter(@PathParam("id") String id);
+
+   @Named("datacenter:get")
+   @GET   
+   @Path("/{id}")
+   @ResponseParser(DataCenterParser.class)
+   @Produces("application/vnd.profitbricks.resource+json")
+   @Fallback(NullOnNotFoundOr404.class)
+   DataCenter getDataCenter(@PathParam("id") String id, DepthOptions options);
+   
+   @Named("datacenter:create")
+   @POST
+   @ResponseParser(DataCenterParser.class)
+   @Produces("application/vnd.profitbricks.resource+json")
+   @MapBinder(DataCenterCreateMapBinder.class)
+   @Fallback(NullOnNotFoundOr404.class)
```

Do not use 404 fallbacks on POST/PUT. Just remove the fallback and let the method throw an exception if a 404 occurs.